src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,191 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from datetime import datetime
+from datetime import timedelta
+import json
+import logging
+import os
+import time
+import sas
+import sas_testcase
+from util import winnforum_testcase
+
+
+class HeartbeatTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_6(self):
+    """Heartbeat Request from CBSD immediately after CBSD's grant is expired.
+
+    Response Code should  be 103.
+    """
+
+    # Register the device.
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    cbsd_ids = self.assertRegistered([device_a])
+
+    # Request grant.
+    grant_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_ids[0]
+    request = {'grantRequest': [grant_0]}
+    # Check grant response.
+    response = self._sas.Grant(request)['grantResponse'][0]
+    self.assertEqual(response['cbsdId'], cbsd_ids[0])
+    self.assertEqual(response['response']['responseCode'], 0)
+    grant_id = response['grantId']
+    grant_expire_time = datetime.strptime(response['grantExpireTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+    #  Test will automatically fail if the SAS UUT is not configured to use a
+    #  short Grant duration to prevent this test from taking unnecessarily
+    #  long to execute.
+    self.assertTrue(grant_expire_time < datetime.utcnow()+timedelta(hours=24))
```

Comment:
Hello Reda,

Thanks for taking the time to review. The decision to configure with shorter grant duration is documented in the testing API doc (Under Section: 'Manual Configuration of SAS under test).

Please feel free to follow-up with the test-dev group, I'll be happy to change this code to reflect any changes. For now, this should be ok to keep and approve?

Reply:
Hello, thank you for your response and for your changes which are ok.
in this section of admin API doc, it is written : Set grant duration to 5 minutes (exact duration TBD)
but I couldn't understand the link with the condition in the code for grant_expire_time to be less than (now + 24 hours).

Reply:
Reda, 
Since a SAS in normal operation would set grant duration to much more than 24 hours,  here we just do a simple check to make sure that it's been reconfigured to "something reasonable" (Could be 5 minutes or 3 hours. Its still not clear and yet to be decided.)
Also, we are OK to let this assertion be lenient -- if we specify 3 hours and someone gives 3 hours + 5 minutes, we don't want to fail them for that. So (now + 24 hours) seems reasonably accommodative.

Reply:
thank you for your explanation, I think we can discuss this later with dev group.